src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -239,10 +238,7 @@ void reconnect() throws TimeoutException, ExecutionException, InterruptedExcepti
                 if (error != null) {
                     // Must synchronize since we're messing with the shared connection object and this block
                     // is executed in some other thread
```

Comment:
can you move/update this comment since the synchronize is inside `connectionCleanup`?

Reply:
Updated